src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkTransition.java

```diff
@@ -445,6 +446,48 @@ public static BuildOptions addDefaultStarlarkOptions(
     return optionsWithDefaults == null ? fromOptions : optionsWithDefaults.build();
   }
 
+  /*
+   * Removes all build settings explicitly set to their default values from the output build options
+   * of a transition. This cancels the effect of addDefaultStarlarkOptions post-transition if called
```

Comment:
Can you elaborate for me this condition? Is there any risk of not calling it on the same buildSettingPackages? Or is this just asserting expected code invariants?